sui_programmability/examples/games_with_chance/sources/satoshi_flip.move

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+
+/// Satoshi flip is a fair and secure way to conduct a coin flip.
+/// A player called "house" chooses a random secret and uploads its hash on chain
+/// Another player called "player",  makes a guess on a predetermined bit (eg. last) of the secret
+/// House reveals the secret and the winner is determined
+/// 
+/// This implementation checks the last bit of the first byte of the secret
+module games_with_chance::satoshi_flip {
+    // imports
+    use std::option::{Self, Option};
+    use std::hash::sha3_256;
+    use std::vector;
+
+    use sui::coin::{Self, Coin};
+    use sui::balance::{Self, Balance};
+    use sui::sui::SUI;
+    use sui::digest::{Self, Sha3256Digest};
+    use sui::object::{Self, UID};
+    use sui::tx_context::{Self, TxContext};
+    use sui::transfer;
+    
+    /* Terminology
+        house: The user who picks the secret
+        player: The user who wages an amount on his guess
+        max_bet: the maximum amount the house is willing to potentialy lose or win
+        min_bet: the minimum amount the house is willing to accept as a wager/stake (too low of an amount might be worth the gas) min_bet <= max_bet
+        stake: the amount the player is willing to wager on his guess, min_bet <= stake <= max_bet
+    */
+
+    /*
+        Conventions:
+        A guess = 2 in the outcome means player has not bet but the game was ended
+        A guess = 3 in the outcome means player has bet but house refused to reveal in time
+    */
+
+    /// Hash function the house will use
+    const HASH: vector<u8> = b"SHA3-256";
```

Comment:
Where is this used? if it's just a comment, just put it in a comment

Reply:
I put it in a constant that in order to be world readable (in explorer, with a call, by another module). I felt it was important since anyone trying to become house should be able to gain this info otherwise (if house chooses a random hash function) the house won't be able to end any games. It's a way of saying "It was written right there that you should use SHA3-256".
The practice is that such info is displayed in the comments?